hangman/src/App.js

```diff
@@ -0,0 +1,223 @@
+      letterStatus[String.fromCharCode(i)] = false
+    }
+    return letterStatus
+  }
+  resetStates = {
+    letterStatus: async () => {
+      let resetLetterStatus = this.generateLetterStatuses()
+      this.setState({ letterStatus: resetLetterStatus })
+    },
+    gameStatus: async () => {
+      let resetGameStatus = this.state.gameStatus
+      resetGameStatus.isGameOver = false
+      resetGameStatus.isPlayerWon = false
+      resetGameStatus.isGameStarted = true
+      this.setState({ gameStatus: resetGameStatus })
+    },
+    score: async () => {
+      this.setState({ score: 100 })
+    },
+  }
+  findSolutionOfLevel = (level) => {
+    const solution = this.state.solutions.find((s) => s.level === level)
+    return solution
+  }
+  endGame = () => {
+    let updatedGameStatus = this.state.gameStatus
+    updatedGameStatus.isGameOver = true
+    updatedGameStatus.isPlayerWon = this.state.score ? true : false
+    this.setState({ gameStatus: updatedGameStatus })
+  }
+  selectLetter = (letter) => {
+    let allLetters = { ...this.state.letterStatus }
+    allLetters[letter] = true
+    this.setState(
+      {
+        letterStatus: allLetters,
+      },
+      () => {
+        const word = this.findSolutionOfLevel(this.state.activeSolution.level)
+          .word
+        let isLetterInWord =
+          word.includes(letter) || word.includes(letter.toLowerCase())
+        if (isLetterInWord) {
+          let updatedSolution = this.state.activeSolution
+          updatedSolution.lettersToFind--
+          this.setState({ activeSolution: updatedSolution }, () => {
+            if (this.state.activeSolution.lettersToFind === 0) {
+              if (
+                this.state.activeSolution.level === this.state.solutions.length
+              ) {
+                this.endGame()
+              }
+            }
+          })
+        } else {
+          const currScore = this.state.score
+          if (currScore > 0) {
+            this.setState({ score: currScore - 10 }, () => {
+              if (this.state.score === 0) {
+                this.endGame()
+              }
+            })
+          }
+        }
+      }
+    )
+  }
+  setActiveSolution = (level) => {
+    const uniqueLettersInWord = []
+    this.findSolutionOfLevel(level)
+      .word.split("")
+      .forEach((letter) => {
+        if (!uniqueLettersInWord.includes(letter.toUpperCase())) {
+          uniqueLettersInWord.push(letter.toUpperCase())
+        }
+      })
+    const lettersToFind = uniqueLettersInWord.length
+    this.setState({
+      activeSolution: { level, lettersToFind },
+    })
+  }
+  getNextLevel = async (level) => {
+    const nextLevel = level + 1
+    await this.setActiveSolution(nextLevel)
+    await this.resetStates.letterStatus()
+  }
+  startGame = async () => {
+    this.setActiveSolution(1)
+    await this.resetStates.gameStatus()
+    await this.resetStates.letterStatus()
+    await this.resetStates.score()
+  }
```

Comment:
There is no synchronize method. don't use `async-await` everywhere...
(setState doesn't return a promise)